### PR TITLE
Use ArgumentNullException.ThrowIfNull and ObjectDisposedException.ThrowIf in more places

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs [hunk 1]
```diff
@@ -523,18 +523,18 @@ internal bool SetInvalidStringFailure()
...
 
-                Debug.Assert(argumentName != null);
-                throw new ArgumentNullException(argumentName, SR.ArgumentNull_String);
+                return false;
             }
 
             internal bool SetOverflowFailure()
             {
-                if (!_throwOnFailure)
+                if (_throwOnFailure)
```
Comment (MihaZupan) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:537]: This one is inverted

**Review by MihaZupan — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs:537]: Oops... not sure how that happened. Thanks.

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CryptoStream.cs [hunk 2]
```diff
@@ -719,8 +719,7 @@ private async Task CopyToAsyncInternal(Stream destination, int bufferSize, Cance
         private void CheckCopyToArguments(Stream destination, int bufferSize)
         {
             ArgumentNullException.ThrowIfNull(destination);
-
-            EnsureNotDisposed(destination, nameof(destination));
+            ObjectDisposedException.ThrowIf(!stream.CanRead && !stream.CanWrite, destination);
```
Comment (stephentoub) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CryptoStream.cs:722]:
```suggestion
            ObjectDisposedException.ThrowIf(!destination.CanRead && !destination.stream.CanWrite, destination);
```

**Comment by marek-safar** [comment]
Are we missing an analyzer for these code patterns or is it just not enabled?

**Comment by stephentoub** [comment]
https://github.com/dotnet/roslyn-analyzers/pull/6293, which is what found/fixed these.

**Pull request merged**